sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -967,6 +967,32 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val FILE_META_CACHE_ENABLED_SOURCE_LIST = buildConf("spark.sql.fileMetaCache.enabledSourceList")
+    .doc("A comma-separated list of data source short names for which data source enabled file " +
+      "meta cache, now the file meta cache only support ORC, it is recommended to enabled this " +
+      "config when multiple queries are performed on the same dataset, default is false." +
+      "Warning: if the fileMetaCache is enabled, the existing data files should not be " +
+      "replaced with the same file name, otherwise there will be a risk of job failure or wrong " +
+      "data reading before the cache entry expires.")
+    .version("3.3.0")
+    .stringConf
```

Comment:
Could you add `.checkValue`? The valid value is only `orc` in this PR.
After merging this PR, you can extend it to `parquet` inside Parquet PR.

Reply:
c3838e6 add `.checkValue` and test case